src/main/kotlin/org/cascadebot/cascadebot/scheduler/ScheduledAction.kt

```diff
@@ -0,0 +1,78 @@
+package org.cascadebot.cascadebot.scheduler
+
+import de.bild.codec.annotations.Id
+import org.bson.types.ObjectId
+import org.cascadebot.cascadebot.data.managers.ScheduledActionManager
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import kotlin.reflect.KClass
+import kotlin.reflect.full.isSubclassOf
+
+data class ScheduledAction(
+        val type: ActionType,
+        val data: ActionData,
+        val guildId: Long,
+        val channelId: Long,
+        val userId: Long,
+        val creationTime: Instant,
+        val executionTime: Instant
+) : Runnable {
+
+    @Id
+    val id: ObjectId = ObjectId()
+
+    val delay: Long
+        get() = ChronoUnit.MILLIS.between(creationTime, executionTime)
```

Comment:
This will cause issues when the node is shut down as its not calculating the delay between now and execution, its doing it between creation and execution. Also this is being stored in mongo when it should be calculated on load.